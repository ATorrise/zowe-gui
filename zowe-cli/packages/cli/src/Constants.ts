/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { ProfileConstants } from "@zowe/core-for-zowe-sdk";

/**
 * Class to contain  constants
 * @export
 * @class Constants
 */
export class Constants {

    /**
     * Custom log location to prevent logging to default folder based on the package name
     * @static
     * @memberof Constants
     */
    public static readonly LOG_LOCATION = "zowe/logs/zowe.log";

    /**
     * Display name
     * @static
     * @memberof Constants
     */
    public static readonly DISPLAY_NAME = "Zowe CLI";

    /**
     * Binary executable name
     * @static
     * @memberof Constants
     */
    public static readonly BINARY_NAME = "zowe";

    /**
     *
     * Documentation link for the CLI
     * @static
     * @memberof Constants
     */
    public static readonly DOCUMENTATION_LINK = "https://docs.zowe.org";

    /**
     *
     * Support link for zowe
     * @static
     * @memberof Constants
     */
    public static readonly SUPPORT_LINK = "https://www.zowe.org";

    /**
     * Description of product
     * @static
     * @memberof Constants
     */
    public static readonly DESCRIPTION = `Welcome to ${Constants.DISPLAY_NAME}!

${Constants.DISPLAY_NAME} is a command line interface (CLI) that provides a simple and streamlined way to interact with IBM z/OS.

For additional ${Constants.DISPLAY_NAME} documentation, visit ${Constants.DOCUMENTATION_LINK}

For ${Constants.DISPLAY_NAME} support, visit ${Constants.SUPPORT_LINK}
`;

    /**
     * Home environment variable
     * @static
     * @memberof Constants
     */
    public static readonly HOME_ENV_KEY = "ZOWE_CLI_HOME";


    /**
     * Prefix for environmental variable settings used by Imperative
     * @static
     * @memberof Constants
     */
    public static readonly ENV_PREFIX = "ZOWE";


    /**
     * Home directory
     * @static
     * @memberof Constants
     */
    public static readonly HOME_DIR = "~/.zowe";

    //______________________________________________________________________
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly BASE_CONNECTION_OPTION_GROUP = ProfileConstants.BASE_CONNECTION_OPTION_GROUP;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly BASE_OPTION_HOST = ProfileConstants.BASE_OPTION_HOST;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly BASE_OPTION_PORT = ProfileConstants.BASE_OPTION_PORT;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly BASE_OPTION_USER = ProfileConstants.BASE_OPTION_USER;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly BASE_OPTION_PASSWORD = ProfileConstants.BASE_OPTION_PASSWORD;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly BASE_OPTION_REJECT_UNAUTHORIZED = ProfileConstants.BASE_OPTION_REJECT_UNAUTHORIZED;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly BASE_OPTION_TOKEN_TYPE = ProfileConstants.BASE_OPTION_TOKEN_TYPE;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly BASE_OPTION_TOKEN_VALUE = ProfileConstants.BASE_OPTION_TOKEN_VALUE;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly BASE_OPTION_CERT_FILE = ProfileConstants.BASE_OPTION_CERT_FILE;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly BASE_OPTION_CERT_KEY_FILE = ProfileConstants.BASE_OPTION_CERT_KEY_FILE;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    // public static readonly BASE_OPTION_CERT_FILE_PASSPHRASE = ProfileConstants.BASE_OPTION_CERT_FILE_PASSPHRASE

    //______________________________________________________________________
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly AUTO_INIT_OPTION_GROUP = ProfileConstants.AUTO_INIT_OPTION_GROUP;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly AUTO_INIT_OPTION_HOST = ProfileConstants.AUTO_INIT_OPTION_HOST;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly AUTO_INIT_OPTION_PORT = ProfileConstants.AUTO_INIT_OPTION_PORT;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly AUTO_INIT_OPTION_USER = ProfileConstants.AUTO_INIT_OPTION_USER;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly AUTO_INIT_OPTION_PASSWORD = ProfileConstants.AUTO_INIT_OPTION_PASSWORD;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly AUTO_INIT_OPTION_REJECT_UNAUTHORIZED = ProfileConstants.AUTO_INIT_OPTION_REJECT_UNAUTHORIZED;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly AUTO_INIT_OPTION_TOKEN_TYPE = ProfileConstants.AUTO_INIT_OPTION_TOKEN_TYPE;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly AUTO_INIT_OPTION_TOKEN_VALUE = ProfileConstants.AUTO_INIT_OPTION_TOKEN_VALUE;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly AUTO_INIT_OPTION_CERT_FILE = ProfileConstants.AUTO_INIT_OPTION_CERT_FILE;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
    */
    public static readonly AUTO_INIT_OPTION_CERT_KEY_FILE = ProfileConstants.AUTO_INIT_OPTION_CERT_KEY_FILE;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly AUTH_GROUP_SUMMARY = ProfileConstants.AUTH_GROUP_SUMMARY;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly AUTH_GROUP_DESCRIPTION = ProfileConstants.AUTH_GROUP_DESCRIPTION;

    //______________________________________________________________________
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly APIML_LOGIN_SUMMARY = ProfileConstants.APIML_LOGIN_SUMMARY;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly APIML_LOGIN_DESCRIPTION = ProfileConstants.APIML_LOGIN_DESCRIPTION;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly APIML_LOGIN_EXAMPLE1 = ProfileConstants.APIML_LOGIN_EXAMPLE1;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly APIML_LOGIN_EXAMPLE2 = ProfileConstants.APIML_LOGIN_EXAMPLE2;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly APIML_LOGOUT_SUMMARY = ProfileConstants.APIML_LOGOUT_SUMMARY;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly APIML_LOGOUT_DESCRIPTION = ProfileConstants.APIML_LOGOUT_DESCRIPTION;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly APIML_LOGOUT_EXAMPLE1 = ProfileConstants.APIML_LOGOUT_EXAMPLE1;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly APIML_LOGOUT_EXAMPLE2 = ProfileConstants.APIML_LOGOUT_EXAMPLE2;
    /**
     * @deprecated Please use this constant from the @zowe/core-for-zowe-sdk
     */
    public static readonly APIML_LOGOUT_OPTION_TOKEN_TYPE = ProfileConstants.APIML_LOGOUT_OPTION_TOKEN_TYPE;
}
