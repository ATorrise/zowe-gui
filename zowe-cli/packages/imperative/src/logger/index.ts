/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

export * from "./src/doc/IConfigLogging";
export * from "./src/doc/ILog4jsAppender";
export * from "./src/doc/ILog4jsConfig";
export * from "./src/doc/ILog4jsLayout";
export * from "./src/Logger";
export * from "./src/LoggerConfigBuilder";
export * from "./src/LoggerUtils";
