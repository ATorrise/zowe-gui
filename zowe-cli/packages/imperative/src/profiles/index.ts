/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

export * from "./src/constants/ProfilesConstants";

export * from "./src/doc";

export * from "./src/doc/definition/IMetaProfile";
export * from "./src/doc/definition/IProfile";
export * from "./src/doc/definition/IProfileDependency";
export * from "./src/doc/definition/IProfileProperty";
export * from "./src/doc/definition/IProfileSchema";
export * from "./src/doc/api/IProfileManagerFactory";

export * from "./src/doc/parms/IDeleteProfile";
export * from "./src/doc/parms/ILoadProfile";
export * from "./src/doc/parms/IProfileManager";
export * from "./src/doc/parms/IProfileManagerInit";
export * from "./src/doc/parms/ISaveProfile";
export * from "./src/doc/parms/ISaveProfileFromCliArgs";
export * from "./src/doc/parms/ISetDefaultProfile";
export * from "./src/doc/parms/IUpdateProfile";
export * from "./src/doc/parms/IUpdateProfileFromCliArgs";
export * from "./src/doc/parms/IValidateProfile";
export * from "./src/doc/parms/IValidateProfileForCLI";
export * from "./src/doc/parms/IValidateProfileWithSchema";

export * from "./src/doc/api/IProfileManagerFactory";

export * from "./src/doc/response/IProfileDeleted";
export * from "./src/doc/response/IProfileInitialized";
export * from "./src/doc/response/IProfileLoaded";
export * from "./src/doc/response/IProfileSaved";
export * from "./src/doc/response/IProfileUpdated";
export * from "./src/doc/response/IProfileValidated";

export * from "./src/utils/ProfileIO";
export * from "./src/utils/ProfileUtils";
export * from "./src/utils";

export * from "./src/validation/api/ProfileValidator";

export * from "./src/validation/doc/IProfileValidationPlan";
export * from "./src/validation/doc/IProfileValidationReport";
export * from "./src/validation/doc/IProfileValidationTask";
export * from "./src/validation/doc/IProfileValidationTaskResult";

export * from "./src/BasicProfileManager";
export * from "./src/BasicProfileManagerFactory";

export * from "./src/abstract/AbstractProfileManagerFactory";
