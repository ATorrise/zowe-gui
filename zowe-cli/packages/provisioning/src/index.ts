/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

export * from "./doc/input/IPropertiesInput";
export * from "./doc/input/IProvisionOptionals";

export * from "./doc/zosmf/IPerformActionResponse";
export * from "./doc/zosmf/IPromptVariable";
export * from "./doc/zosmf/IProvisionedInstance";
export * from "./doc/zosmf/IProvisionedInstanceActions";
export * from "./doc/zosmf/IProvisionedInstances";
export * from "./doc/zosmf/IProvisionedInstanceVariable";
export * from "./doc/zosmf/IProvisionedInstanceVariables";
export * from "./doc/zosmf/IProvisionTemplateResponse";
export * from "./doc/zosmf/IPublishedTemplate";
export * from "./doc/zosmf/IPublishedTemplateInfo";
export * from "./doc/zosmf/IPublishedTemplates";
export * from "./doc/zosmf/ITemplateRegistryInfo";
export * from "./doc/zosmf/ITemplateWorkflowInfo";

export * from "./DeleteInstance";
export * from "./ListCatalogTemplates";
export * from "./ListInstanceInfo";
export * from "./ListInstanceVariables";
export * from "./ListRegistryInstances";
export * from "./ListTemplateInfo";
export * from "./PerformAction";
export * from "./ProvisioningConstants";
export * from "./ProvisioningService";
export * from "./ProvisioningValidator";
export * from "./ProvisionPublishedTemplate";
