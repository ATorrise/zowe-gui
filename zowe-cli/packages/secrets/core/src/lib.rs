pub mod os;
mod tests;