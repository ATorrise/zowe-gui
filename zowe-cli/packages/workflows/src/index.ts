/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

/**
 * Use the index.ts to export any public facing APIs/intefaces/etc.
 *
 * If your plugin introduces a set of APIs/functions that others would find useful when building node apps
 * (or editor extensions, etc.) export them here.
 *
 * For example, Zowe CLI offers Jobs APIs that can be invoke programmatically from a VS code extension to create
 * a Job viewer/tree extension.
 */

export * from "./doc/IActiveWorkflows";
export * from "./doc/IArchivedWorkflow";
export * from "./doc/IArchivedWorkflows";
export * from "./doc/IAutomationStatus";
export * from "./doc/ICreatedWorkflow";
export * from "./doc/ICreatedWorkflowLocal";
export * from "./doc/ICreateWorkflow";
export * from "./doc/IGetWorkflowsOptions";
export * from "./doc/IJobFiles";
export * from "./doc/IJobInfo";
export * from "./doc/IJobStatus";
export * from "./doc/IPropertyMapping";
export * from "./doc/IStartWorkflow";
export * from "./doc/IStepApprovers";
export * from "./doc/IStepDefinition";
export * from "./doc/IStepInfo";
export * from "./doc/IStepSummary";
export * from "./doc/IVariable";
export * from "./doc/IVariableDefinition";
export * from "./doc/IVariableInfo";
export * from "./doc/IVariableSpecification";
export * from "./doc/IWorkflowDefinition";
export * from "./doc/IWorkflowInfo";
export * from "./doc/IWorkflows";
export * from "./doc/IWorkflowsInfo";

export * from "./ArchivedDelete";
export * from "./ArchiveWorkflow";
export * from "./Cancel";
export * from "./Create";
export * from "./Definition";
export * from "./Delete";
export * from "./ListArchivedWorkflows";
export * from "./ListWorkflows";
export * from "./Properties";
export * from "./Start";
export * from "./WorkflowConstants";
export * from "./WorkflowValidator";
