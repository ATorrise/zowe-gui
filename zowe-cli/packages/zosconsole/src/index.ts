/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

export * from "./doc/zosmf/IZosmfCollectResponse";
export * from "./doc/zosmf/IZosmfIssueParms";
export * from "./doc/zosmf/IZosmfIssueResponse";

export * from "./doc/ICollectParms";
export * from "./doc/IConsoleParms";
export * from "./doc/IConsoleResponse";
export * from "./doc/IIssueParms";

export * from "./types/ConsoleResponse";

export * from "./CollectCommand";
export * from "./ConsoleConstants";
export * from "./ConsoleResponseService";
export * from "./ConsoleValidator";
export * from "./IssueCommand";
