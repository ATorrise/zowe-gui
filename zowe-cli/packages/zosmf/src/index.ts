/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

export * from "./constants/Zosmf.constants";
export * from "./constants/Zosmf.messages";
export * from "./constants/Zosmf.profile";

export * from "./doc/IDefinedSystem";
export * from "./doc/IZosmfInfoResponse";
export * from "./doc/IZosmfListDefinedSystemsResponse";
export * from "./doc/IZosmfPluginInfo";

export * from "./CheckStatus";
export * from "./ListDefinedSystems";
export * from "./ZosmfBaseHandler";
export * from "./ZosmfSession";

