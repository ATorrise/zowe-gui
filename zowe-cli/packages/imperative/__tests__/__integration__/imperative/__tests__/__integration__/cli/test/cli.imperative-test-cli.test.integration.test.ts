/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { runCliScript } from "../../../../../../src/TestUtil";
import { ITestEnvironment } from "../../../../../../__src__/environment/doc/response/ITestEnvironment";
import { SetupTestEnvironment } from "../../../../../../__src__/environment/SetupTestEnvironment";

// Test Environment populated in the beforeAll();
let TEST_ENVIRONMENT: ITestEnvironment;

describe("imperative-test-cli test command", () => {
    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENVIRONMENT = await SetupTestEnvironment.createTestEnv({
            cliHomeEnvVar: "IMPERATIVE_TEST_CLI_CLI_HOME",
            testName: "imperative_test_cli_test_command"
        });
    });

    it("should display the help", async () => {
        const response = runCliScript(__dirname + "/__scripts__/test_help.sh", TEST_ENVIRONMENT.workingDir);
        expect(response.status).toBe(0);
        expect(response.stderr.toString()).toBe("");
        expect(response.stdout.toString()).toMatchSnapshot();
    });
});
