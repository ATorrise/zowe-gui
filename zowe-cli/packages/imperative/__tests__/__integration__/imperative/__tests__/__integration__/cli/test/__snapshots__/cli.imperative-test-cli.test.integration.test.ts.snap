// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imperative-test-cli test command should display the help 1`] = `
"
 DESCRIPTION
 -----------

   Test that various imperative features are working

 USAGE
 -----

   imperative-test-cli test <command>

   Where <command> is one of the following:

 COMMANDS
 --------

   logging                Test imperative logging
   masking                Test imperative masking
   test-config-auto-store Read some config       
   test-config-override   Override some config   

 GLOBAL OPTIONS
 --------------

   --help-examples (boolean)

      Display examples for all the commands in a group

   --response-format-json | --rfj (boolean)

      Produce JSON formatted data from a command

   --help | -h (boolean)

      Display help text

   --help-web | --hw (boolean)

      Display HTML help in browser

{
  \\"success\\": true,
  \\"exitCode\\": 0,
  \\"message\\": \\"The help was constructed for command: test.\\",
  \\"stdout\\": \\"\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Test that various imperative features are working\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   imperative-test-cli test <command>\\\\n\\\\n   Where <command> is one of the following:\\\\n\\\\n COMMANDS\\\\n --------\\\\n\\\\n   logging                Test imperative logging\\\\n   masking                Test imperative masking\\\\n   test-config-auto-store Read some config       \\\\n   test-config-override   Override some config   \\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --help-examples (boolean)\\\\n\\\\n      Display examples for all the commands in a group\\\\n\\\\n   --response-format-json | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n   --help-web | --hw (boolean)\\\\n\\\\n      Display HTML help in browser\\\\n\\\\n\\",
  \\"stderr\\": \\"\\",
  \\"data\\": \\"\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Test that various imperative features are working\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   imperative-test-cli test <command>\\\\n\\\\n   Where <command> is one of the following:\\\\n\\\\n COMMANDS\\\\n --------\\\\n\\\\n   logging                Test imperative logging\\\\n   masking                Test imperative masking\\\\n   test-config-auto-store Read some config       \\\\n   test-config-override   Override some config   \\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --help-examples (boolean)\\\\n\\\\n      Display examples for all the commands in a group\\\\n\\\\n   --response-format-json | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n   --help-web | --hw (boolean)\\\\n\\\\n      Display HTML help in browser\\\\n\\\\n\\"
}"
`;
